src/pages/Login/Login.jsx

```diff
@@ -1,5 +1,211 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
 
 export default function Login() {
-  return <div></div>;
+  const [emailValue, setEmailValue] = useState('');
+  const [passwordValue, setPasswordValue] = useState('');
+  const [isEmailValid, setIsEmailValid] = useState('');
+  const [isPasswordValid, setIsPasswordValid] = useState('');
+
+  const controlValue = event => {
+    if (event.target.name === 'inputEmail') {
+      setEmailValue(event.target.value);
+    } else if (event.target.name === 'inputPassword') {
+      setPasswordValue(event.target.value);
+    }
+  };
+
+  const allValueCheck = () => {
+    const reg_email = /^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/;
+
+    const reg_pwd =
+      /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%#?&])[A-Za-z\d@$!%*#?&]{8,16}$/;
+
+    setIsEmailValid(reg_email.test(emailValue));
+    console.log('이메일 정규식 확인', isEmailValid);
+
+    setIsPasswordValid(reg_pwd.test(passwordValue));
+
+    console.log('비밀번호 정규식 확인', isPasswordValid);
+  };
+
+  const goToMain = e => {
+    e.preventDefault();
+
+    if (!isEmailValid && !isPasswordValid) {
+      return;
+    }
+
+    fetch(`http://ambitiouskyle.iptime.org:6389/users/signin`, {
+      method: 'post',
+      body: JSON.stringify({
+        email: emailValue,
+        password: passwordValue,
+      }),
+    })
+      .then(response => response.json())
+      .then(result => {
+        //임의로 작성
+        if (result.message === 'SUCCESS') {
+          localStorage.setItem('TOKEN', result.access_token);
+        } else {
+          alert('잘못된 정보입니다. 다시 입력해주세요!');
+        }
+      });
+  };
+
+  // 두 값 모두 유효할떄 버튼 디스에이블 = false 로  만들기
+  //
+  const isAllValueValid = () => {
+    const isAllValueValid = isEmailValid && isPasswordValid;
+    return isAllValueValid;
+  };
+
+  return (
+    <LoginForm onSubmit={goToMain}>
+      <div className="loginLogo">
+        <img className="logoImage" alt="gream logo" />
+      </div>
+      {/* <div className="loginTitle">로그인</div> */}
+      <div className="inputEmailBox">
+        <InputTitle>이메일 주소</InputTitle>
+        <InputEmail
+          type="text"
+          name="inputEmail"
+          className="inputEmail"
+          placeholder="예) gream@gream.co.kr"
+          onChange={controlValue}
+          onKeyUp={allValueCheck}
+        />
+      </div>
+      <InputPasswordBox>
+        <InputTitle>비밀번호</InputTitle>
+        <InputPassword
+          type="password"
+          name="inputPassword"
+          className="inputPassword"
+          placeholder="비밀번호"
+          onChange={controlValue}
+          onKeyUp={allValueCheck}
+        />
+      </InputPasswordBox>
+      <BasicLoginButton
+        type="submit"
+        className="submitLoginInfoBtn"
+        disabled={!isAllValueValid()} //리턴값 받기가 맞는 방식 ? 색상 인라인으로 안하려면 어떻게 ? (pr) 올리기
```

Comment:
```suggestion
        disabled={!(isEmailValid && isPasswordValid)}
```
- 구현하신 방법대로 하셔도 좋고, 함수가 반복되서 사용하는 게 아니라면, 이메일 값과 패스워드 값 체크 데이터가 이미 있기 때문에 위와 같이 바로 사용하시는 게 좋을 것 같습니다.

Reply:
이미   const [isEmailValid, setIsEmailValid] = useState(false);
  const [isPasswordValid, setIsPasswordValid] = useState(false); 
라고 상단에 정의해 두었으니 굳이 다른 곳에서도 호출할 함수가 아니라면 isEmailValid && isPasswordValid 를 사용하는게 맞다는 말씀이시죠? 적용하겠습니다 !